internal-services/catalog/create-advisory-task.yaml

```diff
@@ -0,0 +1,115 @@
+    - name: application
+      type: string
+      description: Application being released
+  results:
+    - name: result
+      description: Success if the task succeeds, the error otherwise
+  steps:
+    - name: create-advisory
+      image: quay.io/jbieren/release-utils:ansible_5
+      env:
+        - name: GITLAB_HOST
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_host
+        - name: ACCESS_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_access_token
+        - name: GIT_AUTHOR_NAME
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_name
+        - name: GIT_AUTHOR_EMAIL
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_email
+      script: |
+          #!/usr/bin/env sh
+          set -eox pipefail
+
+          REPO_BRANCH=main
+
+          # loading git and gitlab functions
+          . /home/utils/gitlab-functions
+          . /home/utils/git-functions
+          gitlab_init
+          git_functions_init
+
+          function get_advisory_type {
+              NUM_CVES=$(jq '.spec.content.images[].cves.fixed | length' <<< '$(params.advisory_json)' \
+                  | awk '{sum=sum+$0} END{print sum}')
+              if [ $NUM_CVES -gt 0 ] ; then
+                  echo "RHSA"
+                  return
+              fi
+              NUM_ISSUES=$(jq '.spec.issues.fixed | length' <<< '$(params.advisory_json)')
+              if [ $NUM_ISSUES -gt 0 ] ; then
+                  echo "RHBA"
+                  return
+              fi
+              echo "RHEA"
+          }
+
+          function get_advisory_num {
+              # This function will skip many advisory numbers as it is so tied to date. Need to
+              # discover if skipping advisory numbers like this is okay.
+              # There is also a race condition picking the advisory number like this.
+              EXISTING=$(for advisory in $(find data/advisories/*/$(date '+%Y') -name "advisory.yaml") ; do
+                  yq '.metadata.name' $advisory | grep ":$ADVISORY_DAY" || true
+              done | sort -r | head -n 1)
+              if [[ ! -z "$EXISTING" ]] ; then
+                  echo "$((${EXISTING#*:}+1))"
+                  return
+              fi
+              echo "${ADVISORY_DAY}1" # First advisory of the day
+          }
+
+          # This also cds into the git repo
+          git_clone_and_checkout --repository "$(params.repo)" --revision "$REPO_BRANCH"
+
+          PRODUCT_ID=$(jq '.spec.product_id' <<< '$(params.advisory_json)')
+          ADVISORY_DAY=$(date '+%m%d')
+          ADVISORY_DIR="data/advisories/$PRODUCT_ID/$(date '+%Y')" # group advisories by product/year
+          ADVISORY_NUM=$(get_advisory_num)
```

Comment:
same about passing the $ADVISORY_DAY as a parameter.

Reply:
fixed